### ISSUE TITLE
MIDI pitches are incorrect for B# and B##

### ISSUE THREAD

**Issue by newlandsvalley** [newlandsvalley/purescript-abc-parser#40]
We convert an ABC note's pitch  (pitch class, accidental, octave) to a MIDI pitch.  The octave is determined by the  commas attached to the pitch class  in the ABC source,  `(A, A A' etc.)` however we constrain the mapping between ABC and MIDI pitch by assuming that the pitch offset within any given octave must lie between 0 and 11 (because of the 12 note scale).

In fact, this is incorrect for B# and B## because they effectively belong in the next octave up.  i.e. they should have in-octave offsets of 12 and 13 respectively.

We should also take the opportunity to move the MIDI pitch calculating functions to their own module within MIDI.

**Comment by newlandsvalley** [comment]
Fixed by addafb90e42181ffe856e63e5c67e9c4af62f703 in the ps014 branch.

**Comment by jgarte** [comment]
Should this be closed or not until it is merged in?

**Comment by newlandsvalley** [comment]
I've had a problem in that I've only just been able to publish ```abc-parser``` a couple of days ago after a wait for two years because there was no ```rationals``` available.  As soon as the new ```Registry``` exports to the package-sets, I'll close all these issues.

**Comment by newlandsvalley** [comment]
merged into master
